src/api/api.ts

```diff
@@ -0,0 +1,37 @@
+import axios from "axios";
+import { Profile } from "../features/profile/types";
+
+const baseUrl = "/api/";
+
+interface GetProfileResponse {
+  twitchId: string;
+  displayName: string;
+  loginName: string;
+  balance: number;
+  lastTransactionId: number;
+}
+
+export async function fetchProfile(): Promise<Profile> {
+  const response = await axios.get<GetProfileResponse>(baseUrl + "me");
+  var returnValue: Profile;
+
+  if (response.status === 200) {
+    returnValue = {
+      isAuthenticated: true,
+      displayName: response.data.displayName,
+      balance: response.data.balance,
+    } as Profile;
+  } else {
+    returnValue = {
+      isAuthenticated: false,
+      displayName: "",
+      balance: 0,
+    } as Profile;
+  }
+
+  return returnValue;
+}
+
+export async function postPresence() {
+    await axios.post(baseUrl + "presence");
+}
```

Comment:
If you don't need function hoisting, single-line functions have a concise representations if that's your thing

```JavaScript
export const postPresence = () => axios.post(baseUrl + 'presence');

// Axios promises don't always play well with the native Promise API
// you may need to revert the async/await
export const postPresence = async () => await axios.post(baseUrl + 'presence');
```